src/types.jl

```diff
@@ -268,6 +268,21 @@ struct File{T<:SUPPORTED_SAMPLE_TYPES,I<:IO}
     signals::Vector{Union{Signal{T},AnnotationsSignal}}
 end
 
+# allow arbitrary collections (AbstractVector or Tuple)
+# which may or may not contain both Signal and AnnotationsSignal,
+# e.g. Vector{Signal{T}}
+# see https://discourse.julialang.org/t/writing-an-edf-file-using-edf-jl/128666
+function File(io::IO, header::FileHeader, signals)
+    eltypes = unique(typeof.(signals))
+    signal_types = filter(x -> x <: Signal, eltypes)
+    length(signal_types) > 1 &&
+        throw(ArgumentError("Signals must all have the same sample type"))
+    # default to EDF for pure annotation signals
+    signal_type = isempty(signal_types) ? Signal{Int16} : only(signal_types)
+    signals = collect(Union{EDF.AnnotationsSignal, signal_type}, signals)
+    return File(io, header, signals)
+end
```

Comment:
This seems more complex than is necessary? Could you do:
```julia
function File(io::IO, header::FileHeader, signals)
    T = sample_type(header)
    s = convert(Vector{Union{Signal{T},AnnotationsSignal}}, signals)
    return File{T,typeof(io)}(io, header, signals)
end
```
?

Reply:
I did something similar in [bc4c2d3](https://github.com/beacon-biosignals/EDF.jl/pull/96/commits/bc4c2d3caff4f603fd5a1605539756b6c7fe5c4a)

Reply:
I recommend `convert` instead of `collect`, as the former avoids making a copy if the input already has the appropriate type, plus allowing completely arbitrary iterables doesn't seem particularly useful to me.

Reply:
if the input already has the appropriate type, then the default constructor will be invoked instead of this one

Reply:
Ah right, yeah. I thought maybe `convert` avoids a copy when the container element type of the source is a subtype or supertype of the container element type of the destination, but thinking about it more, I'm not actually sure that's true.